### ISSUE TITLE
Simplify browserify bundles with `require` module

### ISSUE THREAD

**Issue by nickpeihl** [cugos/dropchop#263]
I started work over the weekend to `require` modules within the dropchop source. This would be one of the first steps in removing the dependency on Gulp as we could run browserify as a `npm` one-liner rather than streaming each module through Gulp.

**Comment by mapsam** [comment]
Nice! I started taking a peak at the lift this is going to be .. it's more complex than I hoped, considering everything is written in the `dc = dc || {}` syntax. Some thoughts around how to initialize each `*.dropchop.js`'s `init()` function and continue using `$(dc).on` and `$(dc).trigger` as the publication/subscription model would be 👍

**Comment by nickpeihl** [comment]
Speculating here with some pseudocode. Maybe make each piece modular like
```
module.exports = function() {
...
return someInitFunction
}
```

Then in `dropchop.js`
```
var map = require('./dropchop.map')
...
dc.map = map()
...
```

**Comment by wboykinm** [comment]
@mapsam Is this a hard block on #261 or does it not affect modularization?

**Comment by nickpeihl** [comment]
@wboykinm I don't think so. You could probably add each of the turf modules to [this array](https://github.com/cugos/dropchop/blob/master/gulpfile.js#L45) instead of the main turf library. It's worth a try since this issue might take a while.

**Comment by mapsam** [comment]
Indeed - @nickpeihl's array idea should work just fine for now.

**Comment by nickpeihl** [comment]
@mapsam: Do you have any references or links to the type of architecture design used in dropchop? I'm not familiar with this style of application architecture.

**Comment by mapsam** [comment]
@nickpeihl all of dropchop is currently written in the "module pattern" (not to be confused with npm modules) and I always find this article super helpful in how to organize code, globals, and share across files: http://www.adequatelygood.com/JavaScript-Module-Pattern-In-Depth.html

Introducing `require` and browserify may or may not require us to re-architect how dropchop is written.
